### PR TITLE
Fixed weapon pickup and mapstart errors

### PR DESCRIPTION
**Pull request by NovaAstral**
- Fixed error if you picked up a weapon right after spawning it (if you spawned it at your feet)
- Fixed error when map starts

### CODE REVIEW

**Comment by generalwrex** [comment]
Can someone please review this! Then I can merge it!

File: lua/autorun/server/sv_caf_autostart.lua
Comment (X-Coder) [inline lua/autorun/server/sv_caf_autostart.lua:124]: I haven't programmed for garrysmod in a while, but wouldn't this raise a error if ent is really NULL for whatever reason?

OnEntitySpawn is called by garrysmod hooks for PlayerSpawnedNPC, PlayerInitialSpawn, PlayerSpawnedProp, PlayerSpawnedSENT, PlayerSpawnedVehicle

Comment (The17thDoctor) [reply lua/autorun/server/sv_caf_autostart.lua:124]: Why not just replace ent:IsValid() by IsValid(ent) which according to the wiki nilchecks + validchecks

Comment (NovaAstral) [reply lua/autorun/server/sv_caf_autostart.lua:124]: > I haven't programmed for garrysmod in a while, but wouldn't this raise a error if ent is really NULL for whatever reason?

LUA doesn't null, it nil's.
Either way, I will switch it to IsValid(ent)

**Review by The17thDoctor — APPROVED** [review]
Looks good to me now 👍

File: lua/entities/other_screen/init.lua
Comment (X-Coder) [inline lua/entities/other_screen/init.lua:1]: If the screen can only be turned on if there is power, what about if the power is lost later?

Comment (X-Coder) [reply lua/entities/other_screen/init.lua:1]: Maybe the this feature should be put in a separate merge anyway. Because it has nothing to do with these bugfixes.

Comment (The17thDoctor) [reply lua/entities/other_screen/init.lua:1]: If power is lost later, the screen turns off and will need power back to be turned on.

It just fixes the screen turning on for a split second when there's no power as far as I can see.

Comment (X-Coder) [reply lua/entities/other_screen/init.lua:1]: > If power is lost later, the screen turns off and will need power back to be turned on.

If that is already the case, than it is ok and we don't need any changes.

@generalwrex do you want to keep [8d0800e](https://github.com/spacebuild/spacebuild/pull/160/commits/8d0800e0438ceb47240310251885c21d1b4febf3) within this merge request?
If yes - then we can merge this pull request.

Comment (NovaAstral) [reply lua/entities/other_screen/init.lua:1]: > If the screen can only be turned on if there is power, what about if the power is lost later?

it already turns off, my fix (and im going to do it to every other thing in CAF later when its not so hot out), is that you can turn it on for a second or so until the think runs, and turns the screen off because it runs the "are we out of power?" check

**Review by ApertureDevelopment — APPROVED** [review]
Just to also add my 2 cents: Looks good to me. Since everything seems to be talked about, ~~I am gonna approve this PR.~~ Cancel that, I thought I also had write access, but I can't approve this PR. Seems like X-Coder needs to do the honor xD. The only thing to think about is if you want to remove the LS Info screen commit from this PR for clean PR separation.

https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request

**Review by ApertureDevelopment — APPROVED** [review]

**Comment by NovaAstral** [comment]
> The only thing to think about is if you want to remove the LS Info screen commit from this PR for clean PR separation.
> https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request
 
jfc why can't removing a commit from a pr be a simple button press and confirm

**Comment by ApertureDevelopment** [comment]
If everything was easy and working straight away, most developers would be without a job xD

No but seriopusly, idk. Maybe the ghub cli makes things easier ( haven't used it yet x3 )

**Comment by generalwrex** [comment]
I'll be merging it seeing as someone approved it!

**Comment by X-Coder** [comment]
Thank you for your Help [NovaAstral](https://github.com/NovaAstral) and @generalwrex for merging

**Pull request merged**
